### ISSUE TITLE
[Paramount plus]: Ads support for live streams 

### ISSUE THREAD

**Issue by puppinoo** [matthuisman/slyguy.addons#411]
Hi,
Please feel free to delete this issue if it's a duplicate and sorry for that. Basically when a live event has a transiction to an ad stream, it hangs and I have manually restart.
I read around it's because inputstream addon doesn't support discontinuity. Is that really the issue? If so is there a present or soon to be applied solution for that?

Thanks

**Comment by matthuisman** [comment]
yes, IA issue that they are working on
See https://github.com/xbmc/inputstream.adaptive/issues/1071

**Comment by puppinoo** [comment]
@matthuisman 
Hi, not sure you read closed threads but I didnt want to spam with already handled matters.
Is there any progress about this?
The thread you attached has been closed by @Castagna claiming x-discontinuity issue it's fixed now but to me it's never been solved.
The issue can be light with some sports event like soccer where discontinuity is pretty rare but with some other kind like Tennis where ads are sent every 2 games makes the event unwatchable.

Thanks.

**Comment by matthuisman** [comment]
I don't believe it's solved either. You may want to open a new issue over
at inputstream.adaptive

**Comment by puppinoo** [comment]
Hi,
Not sure if you read and it's relevant but since latest kodi 21.1 nightly discontinuity tag with transition between content streams and ads seems to work.
Didn't test much but it worked fine with Serie A events.

**Comment by matthuisman** [comment]
That's great to hear 😊

**Comment by puppinoo** [comment]
Yeah, still testing it and still not perfect.
The transition from live stream to commercial seems quite robust but the opposite, when from the commercial you have to go back to live stream, sometimes hangs kodi or even crashes it.
But I'm confident they will fix it.

**Comment by micheleciondolo** [comment]
@puppinoo (probably we are both italian).. Have you solved in any way??
watching the Champions League is quite upsetting.. The post match show is full of ads and each one blocks the stream totally.

**Comment by puppinoo** [comment]
> @puppinoo (probably we are both italian).. Have you solved in any way?? watching the Champions League is quite upsetting.. The post match show is full of ads and each one blocks the stream totally.

Hi, unfortunately It still has issues. Sometimes I just sideload official US P+ apk on my FTV Stick to solve that. But now it's not convenient anymiee cause in the latest weeks P+ added full 1080p@60fps (before max quality was 720p) and I can consume that just using this plugin.

**Comment by micheleciondolo** [comment]
i'm pretty sure that 1080p60fps is supported by US apk for Android TV. isn't it?
My problem is . I am using a vpn.. And Paramount + app continues changing the quality (you can't set a fixed resolution on the official app).
this is the only reason why I use kodi with the addon.

**Comment by puppinoo** [comment]
> i'm pretty sure that 1080p60fps is supported by US apk for Android TV. isn't it? My problem is . I am using a vpn.. And Paramount + app continues changing the quality (you can't set a fixed resolution on the official app). this is the only reason why I use kodi with the addon.

Maybe it supports that but quality has always been better using addon.

Stream on the app Is adaptive so if It keeps changing that means vpn doesnt guarantee enough bandwidth so It has to scale down.
I had similar problem and actually I solved using a vps I have in US to expose the P+ stream so I can play from that. Btw you can test smartdnsproxy or getflix DNS service. I used them and they used to work fine

**Comment by micheleciondolo** [comment]
The problem is that the Paramount Plus app has an overly aggressive bandwidth control. Since it doesn't allow choosing a fixed resolution, even a minimal bandwidth drop causes an immediate quality change. On the other hand, streaming with Kodi is perfect.

Anyway, I tried SmartDNSProxy as you suggested, but it doesn't seem possible to use either Kodi or the Paramount app on Android TV. However, it does allow access through the Paramount website.

**Comment by micheleciondolo** [comment]
I stand corrected: SmartDNSProxy also works on Kodi if you disable the SlyGuy proxy. Too bad I’ve already paid for Surfshark VPN, whose Smart DNS obviously doesn’t work with Paramount 😂

**Comment by matthuisman** [comment]
Slyguy proxy shouldn't stop smartdnsproxy working. How / where are you setting up smartdnsproxy?

**Comment by puppinoo** [comment]
Not sure with android TV but on FireOs I could set the curated DNS in the wifi configuration or even better globally on my router in dnsmasq conf. I also had issues with Surfshark. Didnt renew the free month cause It didnt work at all with P+ while NordVpn did.
As I told you now Im using a US vps so I dont have to worry about DNS and vpn.

**Comment by micheleciondolo** [comment]
I was setting it in android TV WiFi settings.
Do you mean I should set it in kodi?

**Comment by puppinoo** [comment]
Maybe you already know that but if you are a supporter you can configure the SmartDns directly in SlyGuy Kodi addons
Check this
https://www.matthuisman.nz/2022/06/slyguy-addons-smart-url-actions.html?m=1

**Comment by puppinoo** [comment]
(Amazing feature. Thanks Matt for that)

**Comment by micheleciondolo** [comment]
Cool! . The problem is that I’ve paid for two years of Surfshark, and it seems that Surfshark’s Smart DNS is useless for watching Paramount Plus.

That said, the VPN works pretty well with Kodi! The issue with ads still remains, though. The VPN becomes a problem when using the official Paramount app.

**Comment by matthuisman** [comment]
even easier than using the smart url actions, is you can set a DNS server via the add-on settings -> network now

@micheleciondolo with the device DNS - Kodi and my proxy should still use that DNS server.
So im confused why it wasn't - unless you had actually changed the above addon DNS settings?

**Comment by puppinoo** [comment]
Did they confirm that? Maybe you can open a ticket and they can fix that.
They offered to fix my issue with P+ when I didnt renew. You can try. About
the ads I guess theres nothing we can do until they fix inputstream.

Il dom 8 dic 2024, 21:17 miki ***@***.***> ha scritto:

**Comment by matthuisman** [comment]
have we confirmed its not slyguy proxy causing issue with ads?
i suspect its inputstream adaptive, but before raising an issue there - its best to disable the slyguy proxy and confirm the issue persists

**Comment by micheleciondolo** [comment]
> have we confirmed its not slyguy proxy causing issue with ads? i suspect its inputstream adaptive, but before raising an issue there - its best to disable the slyguy proxy and confirm the issue persists

Yes! Just tried. Confirmed.

**Comment by matthuisman** [comment]
what kodi version? Anyone tried 22?

**Comment by puppinoo** [comment]
Nightly 21.1 here.

Il dom 8 dic 2024, 21:29 Matt Huisman ***@***.***> ha scritto:

**Comment by micheleciondolo** [comment]
> Nightly 21.1 here.

Me too..

**Comment by puppinoo** [comment]
Happening now in real time with Serie A Napoli Lazio

**Comment by micheleciondolo** [comment]
> Happening now in real time with Serie A Napoli Lazio

LOL.. It always happens with Serie a.. And Champions league.. Just wait the 2nd time and reload the stream.

**Comment by puppinoo** [comment]
Yeah, stuck at the end of commercials.

**Comment by matthuisman** [comment]
be interested if someone could see if 22 is any better.
Can be a pain changing major versions though
But that would give you access to the latest IA

**Comment by puppinoo** [comment]
I was able to capture the log of the freeze. Maybe useful?

`2024-12-08 21:36:49.611 T:19278    info <general>: Opening stream: 4001 source: 256
2024-12-08 21:36:49.611 T:19278 warning <general>: AddOnLog: inputstream.adaptive: OnSegmentChanged: Cannot get the stream sample reader
2024-12-08 21:36:50.439 T:19278    info <general>: Creating video codec with codec id: 27
2024-12-08 21:36:50.439 T:19278   error <general>: CDVDVideoCodecAndroidMediaCodec::Open - InstanceGuard locked
2024-12-08 21:36:50.439 T:19278    info <general>: CVideoPlayerVideo::OpenStream - could not open video codec
2024-12-08 21:36:50.440 T:19278    info <general>: Opening stream: 4002 source: 256
2024-12-08 21:36:50.997 T:19365    info <general>: CVideoPlayerAudio::Process - stream stalled
2024-12-08 21:36:51.361 T:19278    info <general>: Finding audio codec for: 86018
2024-12-08 21:36:51.362 T:19278    info <general>: CDVDAudioCodecAndroidMediaCodec: extradata required for aac decoder!
2024-12-08 21:36:51.362 T:19278    info <general>: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder aac
2024-12-08 21:37:20.917 T:19278    info <general>: Opening stream: 5001 source: 256
2024-12-08 21:37:20.917 T:19278 warning <general>: AddOnLog: inputstream.adaptive: OnSegmentChanged: Cannot get the stream sample reader
2024-12-08 21:37:21.118 T:19365    info <general>: CVideoPlayerAudio::Process - stream stalled
2024-12-08 21:37:22.115 T:19278    info <general>: Creating video codec with codec id: 27
2024-12-08 21:37:22.115 T:19278   error <general>: CDVDVideoCodecAndroidMediaCodec::Open - InstanceGuard locked
2024-12-08 21:37:22.115 T:19278    info <general>: CVideoPlayerVideo::OpenStream - could not open video codec
2024-12-08 21:37:22.115 T:19278    info <general>: Opening stream: 5002 source: 256
2024-12-08 21:37:22.844 T:19278    info <general>: Finding audio codec for: 86018
2024-12-08 21:37:22.844 T:19278    info <general>: CDVDAudioCodecAndroidMediaCodec: extradata required for aac decoder!
2024-12-08 21:37:22.846 T:19278    info <general>: CDVDAudioCodecFFmpeg::Open() Successful opened audio decoder aac
2024-12-08 21:37:22.848 T:19365    info <general>: CActiveAEStream::AddData - messy timestamps, increasing interval for measuring average error to 6000 ms
2024-12-08 21:37:39.542 T:19286   error <general>: AddOnLog: inputstream.adaptive: ParseChildManifest: Period of discontinuity 3 not found, attempt to advance to the next
2024-12-08 21:37:46.825 T:19365    info <general>: CVideoPlayerAudio::Process - stream stalled`

**Comment by puppinoo** [comment]
for the life of me I dunno how to paste proper indented code. Sorry.

**Comment by puppinoo** [comment]
In the next days Ill backup my settings and test kodi 22. Last time if was a mess and had to go back to 21.1 nightly.

**Comment by micheleciondolo** [comment]
> be interested if someone could see if 22 is any better. Can be a pain changing major versions though But that would give you access to the latest IA

I'll try in an hour.. I am pretty sure it's the same.

**Comment by micheleciondolo** [comment]
> even easier than using the smart url actions, is you can set a DNS server via the add-on settings -> network now
> 
> @micheleciondolo with the device DNS - Kodi and my proxy should still use that DNS server. So im confused why it wasn't - unless you had actually changed the above addon DNS settings?

I just became a supporter with a small ko-fi donation.
i'll give a try with surfshark smart dns and maybe with the smartDnsProxy.

**Comment by micheleciondolo** [comment]
@matthuisman  sadly Surfshark Smart DNS doesn't work in any way.. I'll try a refund and I'll switch to SmartDNSProxy.
BTW , i confirm the same ads issues in Kodi 22 (which i'm using right now).

**Comment by puppinoo** [comment]
> @matthuisman sadly Surfshark Smart DNS doesn't work in any way.. I'll try a refund and I'll switch to SmartDNSProxy. BTW , i confirm the same ads issues in Kodi 22 (which i'm using right now).

I suggest you to try to open a ticket explaining the issue.
Maybe they can fix that. It's probably just (just meant -> very complexing) a routing problem.

**Comment by micheleciondolo** [comment]
thank you! now is 59$!

@matthuisman  I noticed that disabling inputstream adaptive i'm able to watch a lot of channels (using ffmpeg i think)..  and the ads are just fine.. Because they're not played at all..  You can wait the stream to resume.. without any crash or issue.
On the other hand.. this is not possible with soccer streams.
Maybe a stupid question but.. Is there a way to not use inputstream adaptive at all?
I'm just watching Arsenal - Newcastle and Kodi crashed brutally at the end of the first time :P

**Comment by matthuisman** [comment]
there is a setting in the addon under playback i think "use inputstream adaptive for hls streams"
Turn that off.

However, some streams REQUIRE IA.
But with that disabled - normal HLS should just go via ffmpeg

**Comment by micheleciondolo** [comment]
> there is a setting in the addon under playback i think "use inputstream adaptive for hls streams" Turn that off.
> 
> However, some streams REQUIRE IA. But with that disabled - normal HLS should just go via ffmpeg

yeah! thank you.. but already tried :) it's the same as disabling it! 
hope they'll fix it in any way in the future !

**Comment by matthuisman** [comment]
i might be able to strip x-disc in my proxy . maybe then IA would be happier...
Not sure though.

Its very annoying!

**Comment by micheleciondolo** [comment]
> i might be able to strip x-disc in my proxy . maybe then IA would be happier... Not sure though.
> 
> Its very annoying!

ok thank you :)
one last thing. If you try the latest kodi dev builds... (with the latest IA and Sly addons)  it seems like .. Soccer Matches don't start at all. You have to try multiple times. I don't know why.

**Comment by matthuisman** [comment]
that would probably be an inputstream adaptive issue then if thats the only thing changing

**Comment by puppinoo** [comment]
Hey,
Not sure if today's inputstream adaptive update did the trick or any of your updates did but I'm watching Serie A match and I watched the half match break with all advs and interviews without any issue.
I Hope it's not Just a fluke.

**Comment by micheleciondolo** [comment]
Hi!
I'll try tomorrow with some serie a match.
Are you sure it works between the 1st and 2nd time??
My biggest nightmare was the champions league post match show..(I suggest it to you ... it's hilarious! with thierry Henry and Micah Richards the others)..Every ad was a crash! Hope it works now :)
sabato 18 gennaio 2025 21:51, puppinoo ***@***.***> ha scritto:

**Comment by matthuisman** [comment]
What Kodi version?

**Comment by micheleciondolo** [comment]
kodi 21.1.
but.. Just crashed!!!!! Atalanta- Napoli.. post match show..
Updated IA (21.5.9)

Inviato con l'email sicura [Proton Mail](https://proton.me/mail/home).

sabato 18 gennaio 2025 23:12, Matt Huisman ***@***.***> ha scritto:

> What Kodi version?
>

**Comment by matthuisman** [comment]
Makes sense I don't think 21 IA will get any more meaningful updates. You
could check it's change log on GitHub

**Comment by puppinoo** [comment]
No,
I'm sorry but it's not true.  It was a mistake my side. Basically I was watching using my vps server instead of Kodi. When I realized that it was too late.
I deleted the comment but I'd probably just edit it.

**Comment by matthuisman** [comment]
i suggest someone raise an issue on iptv simple and provide some sort of access for the dev to test with :)
that takes a bit of time and commitment but he is pretty good once he can confirm the issue etc

Make sure your already testing with Kodi 22 and latest IA or he will ignore it.
Provide the manifests (setting in IA to save them) and a full kodi debug log showing the issue

**Comment by micheleciondolo** [comment]
> Hi guys [@puppinoo](https://github.com/puppinoo) [@micheleciondolo](https://github.com/micheleciondolo) (another ita user to the group here :D ). Any updates on this? I am using the kodi addon for the first time this week and I noticed this well known issue.
> 
> Please let me know if you found a way to let it works.
> 
> Thanks

Hi!
I switched to the official app to avoid this issue.. Sadly I hadn't time to debug it.
